python/cudf/cudf/core/dtypes.py

```diff
@@ -304,3 +305,32 @@ def _validate(cls, precision, scale=0):
             )
         if abs(scale) > precision:
             raise ValueError(f"scale={scale} exceeds precision={precision}")
+
+
+class IntervalDtype(StructDtype):
+    name = "interval"
+
+    def __init__(self, subtype, closed="right"):
+        """
+        subtype: str, np.dtype
+            The dtype of the Interval bounds.
```

Comment:
Since we're capturing `closed` in the dtype, could we update the docstring to capture that?